src/sqlitestorage.cpp

```diff
@@ -368,227 +146,62 @@ bool SqliteStorage::load(const QString &uid)
-        qsize1 = sizeof(SELECT_COMPONENTS_BY_NOTEBOOKUID);
-
-        SL3_prepare_v2(d->mDatabase, query1, qsize1, &stmt1, NULL);
-        u = notebookUid.toUtf8();
-        SL3_bind_text(stmt1, index, u.constData(), u.length(), SQLITE_STATIC);
-
-        count = d->loadIncidences(stmt1);
-    }
-error:
-    d->mIsLoading = false;
-
-    return count >= 0;
-}
-
-bool SqliteStorage::Private::loadRecurringIncidences()
-{
-    if (!mDatabase) {
-        return false;
-    }
-
-    if (mStorage->isRecurrenceLoaded()) {
-        return true;
-    }
-
-    int rv = 0;
-    int count = 0;
-    mIsLoading = true;
-
-    const char *query1 = NULL;
-    int qsize1 = 0;
-
-    sqlite3_stmt *stmt1 = NULL;
-
-    query1 = SELECT_COMPONENTS_BY_RECURSIVE;
-    qsize1 = sizeof(SELECT_COMPONENTS_BY_RECURSIVE);
-
-    SL3_prepare_v2(mDatabase, query1, qsize1, &stmt1, NULL);
-
-    count = loadIncidences(stmt1);
-
-error:
-    mIsLoading = false;
-
-    mStorage->setIsRecurrenceLoaded(count >= 0);
-
-    return count >= 0;
+    Incidence::List list;
+    return d->mBackend.incidences(&list, notebookUid) && d->addIncidences(list, notebookUid);
 }
 
 bool SqliteStorage::search(const QString &key, QStringList *identifiers, int limit)
 {
-    if (!d->mDatabase || key.isEmpty())
-        return false;
-
-    d->mIsLoading = true;
-    const char *query1 = SEARCH_COMPONENTS;
-    int qsize1 = sizeof(SEARCH_COMPONENTS);
-    const QByteArray s('%' + key.toUtf8().replace("\\", "\\\\").replace("%", "\\%").replace("_", "\\_") + '%');
-    int rv = 0;
-    sqlite3_stmt *stmt1 = NULL;
-    int index = 1;
-    Incidence::Ptr incidence;
-    QString nbook;
-    int count = -1;
-
-    qCDebug(lcMkcal) << "Searching DB for" << s;
-    SL3_prepare_v2(d->mDatabase, query1, qsize1, &stmt1, nullptr);
-    SL3_bind_text(stmt1, index, s.constData(), s.length(), SQLITE_STATIC);
-    SL3_bind_text(stmt1, index, s.constData(), s.length(), SQLITE_STATIC);
-    SL3_bind_text(stmt1, index, s.constData(), s.length(), SQLITE_STATIC);
-
-    count = d->loadIncidencesBySeries(stmt1, identifiers, limit);
-
-error:
-    d->mIsLoading = false;
-
-    return count >= 0;
-}
-
-//@cond PRIVATE
-static bool isContaining(const QMultiHash<QString, Incidence::Ptr> &list, const Incidence::Ptr &incidence)
-{
-    QMultiHash<QString, Incidence::Ptr>::ConstIterator it = list.find(incidence->uid());
-    for (; it != list.constEnd(); ++it) {
-        if ((*it)->recurrenceId() == incidence->recurrenceId()) {
-            return true;
+    QHash<QString, Incidence::List> hash;
+    QHash<QString, QStringList> ids;
+    if (d->mBackend.search(&hash, &ids, key, limit) && d->addIncidences(hash)) {
+        if (identifiers) {
+            for (QHash<QString, QStringList>::ConstIterator it = ids.constBegin();
+                 it != ids.constEnd(); it++) {
+                identifiers->append(it.value());
+            }
         }
+        return true;
+    } else {
+        return false;
```

Comment:
Minor, but could drop the else branch and just return false.

Reply:
Sure, done.